src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,105 @@ private void addListingToBusinesses(PlacesSearchResult place) {
     allBusinesses.add(new Listing(name, formattedAddress, 
           placeLocation, rating, photos, types));
   }
-}
\ No newline at end of file
+
+  public void checkNumberOfLocationsOfBusiness() {
+    GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    businesses =  allBusinesses.iterator(); 
+    
+    while (businesses.hasNext()) {
```

Comment:
The problem with doing a for each loop with the linked list, is if I try to remove a an element from the list then I will get the ConcurrentModificationException error. To not get this error I have to use a while loop, using an iterator with it. Also, I have to call .next() before I can use the .remove method for an iterator. This can be found here java67.com/2018/12/how-to-remove-objects-or-elements-while-iterating-Arraylist-java.html.

Reply:
Ok. So it's more than just iteration. In that case you can do a traditional for loop, something like:

```
totalNumberOfBusinesses = allBusinesses.size();
for (for i = 0; i < totalNumberOfBusiness; i++) {
    ...
    if (businessToRemove == true) {
        allBusinesses.remove(i);
    }
    ...
}

Reply:
I keep getting this error (java.util.ConcurrentModificationException), when I try to remove a element from a list inside a for loop or while loop and the only way around that was using an iterator. Instead of removing the big business when it is found, I will recall the remove filterBySmallBusinesses method inside the SmallCityService class after all the big businesses have been added.

Reply:
Also I stop the while loop after fifteen small business have been found, as that is the only thing that displays on the screen.

Reply:
Further thought. I think you're right. The snippet above is invalid.